crates/core_simd/src/ops.rs

```diff
@@ -0,0 +1,624 @@
+    { impl Shr for $type:ty, $scalar:ty } => {
+        impl_op! { @binary $type, $scalar, Shr::shr, ShrAssign::shr_assign, simd_shr }
+    };
+    { impl BitAnd for $type:ty, $scalar:ty } => {
+        impl_op! { @binary $type, $scalar, BitAnd::bitand, BitAndAssign::bitand_assign, simd_and }
+    };
+    { impl BitOr for $type:ty, $scalar:ty } => {
+        impl_op! { @binary $type, $scalar, BitOr::bitor, BitOrAssign::bitor_assign, simd_or }
+    };
+    { impl BitXor for $type:ty, $scalar:ty } => {
+        impl_op! { @binary $type, $scalar, BitXor::bitxor, BitXorAssign::bitxor_assign, simd_xor }
+    };
+
+    { impl Not for $type:ty } => {
+        impl_ref_ops! {
+            impl core::ops::Not for $type {
+                type Output = Self;
+                fn not(self) -> Self::Output {
+                    self ^ !<$type>::default()
+                }
+            }
+        }
+    };
+
+    // generic binary op with assignment when output is `Self`
+    { @binary $type:ty, $scalar:ty, $trait:ident :: $trait_fn:ident, $assign_trait:ident :: $assign_trait_fn:ident, $intrinsic:ident } => {
+        impl_ref_ops! {
+            impl core::ops::$trait<$type> for $type {
+                type Output = $type;
+
+                #[inline]
+                fn $trait_fn(self, rhs: $type) -> Self::Output {
+                    unsafe {
+                        crate::intrinsics::$intrinsic(self, rhs)
+                    }
+                }
+            }
+        }
+
+        impl_ref_ops! {
+            impl core::ops::$trait<$scalar> for $type {
+                type Output = $type;
+
+                #[inline]
+                fn $trait_fn(self, rhs: $scalar) -> Self::Output {
+                    core::ops::$trait::$trait_fn(self, <$type>::splat(rhs))
+                }
+            }
+        }
+
+        impl_ref_ops! {
+            impl core::ops::$trait<$type> for $scalar {
+                type Output = $type;
+
+                #[inline]
+                fn $trait_fn(self, rhs: $type) -> Self::Output {
+                    core::ops::$trait::$trait_fn(<$type>::splat(self), rhs)
+                }
+            }
+        }
+
+        impl_ref_ops! {
+            impl core::ops::$assign_trait<$type> for $type {
+                #[inline]
+                fn $assign_trait_fn(&mut self, rhs: $type) {
+                    unsafe {
+                        *self = crate::intrinsics::$intrinsic(*self, rhs);
+                    }
+                }
+            }
+        }
+
+        impl_ref_ops! {
+            impl core::ops::$assign_trait<$scalar> for $type {
+                #[inline]
+                fn $assign_trait_fn(&mut self, rhs: $scalar) {
+                    core::ops::$assign_trait::$assign_trait_fn(self, <$type>::splat(rhs));
+                }
+            }
+        }
+    };
+}
+
+/// Implements relevant operators for the provided types.
+macro_rules! impl_op_meta {
+    { float: $($scalar:ty => $($vector:ty),*;)* } => {
```

Comment:
Using a magic word inside the macro which must be given when the macro is called and causes it to branch to a different logical implementation suggests that what is desired is to simply use a different name.

Reply:
Looks good!